test/e2e_node/resource_metrics_test.go

```diff
@@ -0,0 +1,134 @@
+
+			matchV1alpha1Expectations := gstruct.MatchAllKeys(gstruct.Keys{
+				"scrape_error": gstruct.Ignore(),
+				"node_cpu_usage_seconds_total": gstruct.MatchAllElements(nodeId, gstruct.Elements{
+					"": boundedSample(1, 1E6),
+				}),
+				"node_memory_working_set_bytes": gstruct.MatchAllElements(nodeId, gstruct.Elements{
+					"": boundedSample(10*framework.Mb, memoryLimit),
+				}),
+
+				"container_cpu_usage_seconds_total": gstruct.MatchElements(containerId, gstruct.IgnoreExtras, gstruct.Elements{
+					fmt.Sprintf("%s::%s::%s", f.Namespace.Name, pod0, "busybox-container"): boundedSample(0, 100),
+					fmt.Sprintf("%s::%s::%s", f.Namespace.Name, pod1, "busybox-container"): boundedSample(0, 100),
+				}),
+
+				"container_memory_working_set_bytes": gstruct.MatchAllElements(containerId, gstruct.Elements{
+					fmt.Sprintf("%s::%s::%s", f.Namespace.Name, pod0, "busybox-container"): boundedSample(10*framework.Kb, 80*framework.Mb),
+					fmt.Sprintf("%s::%s::%s", f.Namespace.Name, pod1, "busybox-container"): boundedSample(10*framework.Kb, 80*framework.Mb),
+				}),
+			})
+			By("Giving pods a minute to start up and produce metrics")
+			Eventually(getV1alpha1ResourceMetrics, 1*time.Minute, 15*time.Second).Should(matchV1alpha1Expectations)
+			By("Ensuring the metrics match the expectations a few more times")
+			Consistently(getV1alpha1ResourceMetrics, 1*time.Minute, 15*time.Second).Should(matchV1alpha1Expectations)
+		})
+		AfterEach(func() {
+			By("Deleting test pods")
+			f.PodClient().DeleteSync(pod0, &metav1.DeleteOptions{}, 10*time.Minute)
+			f.PodClient().DeleteSync(pod1, &metav1.DeleteOptions{}, 10*time.Minute)
+			if !CurrentGinkgoTestDescription().Failed {
+				return
+			}
+			if framework.TestContext.DumpLogsOnFailure {
+				framework.LogFailedContainers(f.ClientSet, f.Namespace.Name, framework.Logf)
+			}
+			By("Recording processes in system cgroups")
+			recordSystemCgroupProcesses()
+		})
+	})
+})
+
+func getV1alpha1ResourceMetrics() (metrics.KubeletMetrics, error) {
+	return metrics.GrabKubeletMetricsWithoutProxy(framework.TestContext.NodeName+":10255", "/metrics/resource/"+v1alpha1.Version)
+}
+
+func nodeId(element interface{}) string {
+	return ""
+}
+
+func containerId(element interface{}) string {
+	el := element.(*model.Sample)
+	return fmt.Sprintf("%s::%s::%s", el.Metric["namespace"], el.Metric["pod"], el.Metric["container"])
+}
+
+func boundedSample(lower, upper interface{}) types.GomegaMatcher {
+	return gstruct.PointTo(gstruct.MatchAllFields(gstruct.Fields{
+		// We already check Metric when matching the Id
+		"Metric": gstruct.Ignore(),
+		"Value":  And(BeNumerically(">=", lower), BeNumerically("<=", upper)),
+		"Timestamp": And(
+			BeNumerically(">=", time.Now().Add(-maxStatsAge).UnixNano()/int64(time.Millisecond)),
+			// Now() is the test start time, not the match time, so permit a few extra minutes.
+			BeNumerically("<", time.Now().Add(2*time.Minute).UnixNano()/int64(time.Millisecond))),
```

Comment:
There's also [`BeTemporally`](https://godoc.org/github.com/onsi/gomega#BeTemporally), not sure if that makes this any cleaner...

Reply:
It is longer to write out, but I think it is a bit clearer.